src/plugins/plugin.chapters.js

```diff
@@ -190,20 +190,14 @@ BookReader.prototype.getOpenLibraryRecord = function () {
     }
   };
 
-  $.ajax({ url: fetchUrlByBookId, dataType: 'jsonp' })
-    .then(data => {
-      if (data && data.length > 0) {
-        return data;
-      } else {
-      // try sourceid
-        return $.ajax({ url: `${baseURL}&source_records=ia:${this.bookId}`, dataType: 'jsonp' });
-      }
-    })
-    .then(data => {
-      if (data && data.length > 0) {
-        setUpChapterMarkers(data[0]);
-      }
-    });
+  const data = await $.ajax({ url: fetchUrlByBookId, dataType: 'jsonp' });
```

Comment:
The logic here isn't quite right. It needs to request the first url to get `data`; and if that doesn't return `data`, it fetches the second request to get `data`. So both `$.ajax` calls should be before the `setUpChapterMarkers` call.